modules/switches.py

```diff
@@ -0,0 +1,107 @@
+
+    def __init__(self, bomb, ident):
+        super().__init__(bomb, ident)
+
+        self.solution = self.gen_rand()
+        while not self.is_valid_state(self.solution):
+            self.solution = self.gen_rand()
+
+        self.switches = self.gen_rand()
+        while True:
+            if self.is_valid_state(self.switches) and self.switches != self.solution:
+                break
+            self.switches = self.gen_rand()
+
+        init_str=" ".join(map(lambda x:("UP","DOWN")[x],self.switches))
+        sltn_str=" ".join(map(lambda x:("UP","DOWN")[x],self.solution))
+        self.log(f"Initial position: {init_str}")
+        self.log(f"Solution: {sltn_str}")
+
+    def generateSwitch(self, index):
+        position = list(map(lambda x: [x[0] + 58 * index, x[1]], self.positions_for_svg_generation))
+        if self.switches[index] == self.D.UP:
+            position[2][1] = 348 - position[2][1]
+            position[3][1] = 348 - position[3][1]
+
+        position = map(lambda x: map(lambda y: str(y), x), position)
+
+        o = f'<circle stroke-width="2" stroke="#000" cx="{58*(index+1)}" cy="174" r="15" />'
+        o += f'<circle stroke-width="2" stroke="#000" fill="{"#0f0" if self.solution[index] == self.D.UP else "#fff"}" cx="{58*(index+1)}" cy="87" r="10" />'
+        o += f'<circle stroke-width="2" stroke="#000" fill="{"#0f0" if self.solution[index] == self.D.DOWN else "#fff"}" cx="{58*(index+1)}" cy="261" r="10" />'
+        o += f'<polygon points="{" ".join(map(lambda x:",".join(x),position))}" style="fill:black;stroke:black;stroke-width:2" />'
+
+        return o
+
+    def get_svg(self, led):
+        output = (f'<svg viewBox="0 0 348 348" fill="#fff" stroke-linecap="butt" stroke-linejoin="round" stroke-miterlimit="10" xmlns:xlink="http://www.w3.org/1999/xlink">'
+                f'<path stroke="#000" stroke-width="2" d="M5 5h338v338h-338z"/>'
+                f'<circle fill="{led}" stroke="#000" cx="298" cy="40.5" r="15" stroke-width="2"/>')
+
+        for i in range(5):
+            output += self.generateSwitch(i)
+        output += '</svg>'
+        return output
+
+    def flip_switch_by_index(self, index):
+        self.switches[index] = not self.switches[index]
+
+    @modules.check_solve_cmd
+    async def cmd_flip(self, author, parts):
+        if not parts:
+            return await self.usage(author)
+
+        input_ = []
+
+        for part in parts:
+            if part.isdigit():
+                for digit in part:
+                    switch = int(digit)
+                    if switch not in range(1, 6):
+                        return await self.bomb.channel.send(f"{author.mention} There is no switch {switch}! There are five switches: 1-5.")
+                    input_.append(switch - 1)
+            else:
+                return await self.bomb.channel.send(f"{author.mention} No such switch: `{part}`")
+
+        while input_ and not self.solved:
```

Comment:
As far as I can see, this `self.solved` condition can never be triggered, because you only set it while calling `handle_solve` below the loop.

Reply:
I some how just completely forgot to add that condition